float32 distance
float64 angle_min
float64 angle_max